common/lib/common-definitions/src/logger.ts

```diff
@@ -59,6 +59,47 @@ export interface ITelemetryPerformanceEvent extends ITelemetryGenericEvent {
     duration?: number; // Duration of event (optional)
 }
 
+/**
+ * Broad classifications to be applied to individual properties as they're prepared to be logged to telemetry.
+ * Please do not modify existing entries for backwards compatibility.
+ */
+ export enum TelemetryDataTag {
+    /** Data containing terms from code packages that may have been dynamically loaded */
```

Comment:
I think the intention (with a move) is to freeze API shape - something that we should definitely do.
Before that happens, should we make it future proof (as much as we can), by exploring what others (like Aria) are doing in same space and making it more inclusive?
For example, PackageData can be generalized to be CodeArtifact, and such category will include things like Stack (as it has artifacts of code - function names).

Reply:
Well the intention is to pave the way to update `ITelemetryBaseLogger` to speak about taggable props so hosts can handle the tagged data themselves.  Perhaps we will adjust these classifications - I have a to-do to reach out to MS Office Telemetry contact about the precedent set for Office Add-ins.  But anyway we already have code using the existing names, so even if we go for a different set, whenever we move to common-def we'll need the full list even deprecated ones.

I'm going to dive in on this in June, but would like to have the framework in place sooner than later, so I think this is good to check in as-is.  @vladsud does that make sense or am I off here?  Related:  https://github.com/microsoft/FluidFramework/issues/5544#issuecomment-827943994

Reply:
We should consider it to be not a string type, but a string.

Consider scenario where our (FF) logger is sandwiched between 3rd party DDS / data store that uses 3rd party tags (and their own implementation of logger extension) and 3rd party host that handles those tags.
This feels similar to extensibility of errorCodes, where each layer is free to create their error codes and all we promise that it's a string, and if you know what version of driver / loader / runtime you use then you know full set, otherwise you do not and should expect unknowns.

I'd say the difference with using Aria (or any other end system) is that they dictate tag usage because it needs to handle them appropriately. Here, we do not know target engine, we also do not control the top of the chain (3rd party may want to build their own logger implementation and use their own tags).

So I feel like we need to be clear about separation of concerns - ITelemetryBaseLogger should use string and not be opinionated about tags, and that (for most part) should be the interface we pass across layers.
On the other hand, ITelemetryLogger is our internal more "powerful" (but also opinionated) interface that instills our own policy including tags and that interface possibly should move out definitions package into telemetry-utils package and usage of this interface should not cross any of the compat boundaries we test (driver/loader/runtime/data store), similar how we should deal with any other "utils" package - feel free to use across layers, but please never ever add it to compat boundary.

What do you think?

Reply:
Ok. Yeah I can get behind tag being any string, it seems appropriately non-opinionated.  Msft's 1P logger can import the tags we come up with in Telemetry-Utils (which will aim to be suitably generic), and if/when we are loading 3P code in 1P app, we must be clear in docs that anything tagged with a custom tag will be redacted in Msft's logging - sounds like a feature.

cc @tylerbutler.

@ksbrar - I think the only change here then is to remove this enum from this PR and change to `tag: string` in the interface.

Reply:
Note - we should validate what I dreamed about with people who drive 3p. Adding @skylerjokiel as well, and would be great to hear from Tyler too. Maybe I'm making it too complex and we will not allow such flexibility - I do not know.

Reply:
I do not think privacy folks would be in position to help here, as the question is - what is useful to 3rd parties. We are not controlling rules here, we only provide framework such that others (2nd party, 3rd party) can achieve their goals.